### PR TITLE
Update payment-handler IDL and tests

### PR DESCRIPTION
**Pull request by lukebjerring**
Hello reviewer(s),

This PR is intended to consolidate the spec’s IDL definition with the WPT test suite’s copy, and any idlharness tests.

The up-to-date copy of the IDL file was automatically extracted from the reffy-reports repo (https://github.com/tidoust/reffy-reports/tree/master/whatwg/idl) which scrapes known specs automatically + regulary.

This PR is part of a migration project which will eventually be automatically updating and creating PRs for changes in spec IDL.

Please check that:
The spec (and its source) is correct and up-to-date
All tests which cover the IDL in the spec have been migrated to fetch + use the idl in the `interfaces/` directory (instead of inline copies in the test files).

### CODE REVIEW

**Comment by lukebjerring** [comment]
@marcoscaceres  - I've taken a stab at adding some instance objects to the test, but I'm not too familiar with `ServiceWorker` and the registration process.

Is there a way of getting a `ServiceWorkerRegistration` object from a SharedWorker and a DedicatedWorker? 
Also, `PaymentManager` doesn't seem to be available on `self` for _any_ of the workers; is that expected current-state? (that's in Chrome Version 68.0.3440.42 (Official Build) beta (64-bit))

**Comment by lukebjerring** [comment]
I think I accidentally 'added' an extra test, instead of renaming the original test, in #9796.

File: interfaces/payment-handler.idl [hunk 1]
```diff
@@ -4,83 +4,87 @@
 // See: https://w3c.github.io/payment-handler/
 
 partial interface ServiceWorkerRegistration {
-    [SameObject]
-    readonly attribute PaymentManager paymentManager;
+  [SameObject] readonly attribute PaymentManager paymentManager;
```
Comment (foolip) [inline interfaces/payment-handler.idl]: Changes here confirmed to be all whitespace with `git diff -w --word-diff-regex=.`

**Review by foolip — APPROVED** [review]
IDL changes LGTM, updated the test enough that it requires review from you, @lukebjerring

**Pull request merged**